boards/arm/tlsr82/tlsr8278adk80d/src/tlsr8278_gpio.c

```diff
@@ -0,0 +1,296 @@
+/****************************************************************************
+ * boards/arm/tlsr82/tlsr8278adk80d/src/tlsr8278_gpio.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdbool.h>
+#include <assert.h>
+#include <errno.h>
+#include <debug.h>
+
+#include <nuttx/clock.h>
+#include <nuttx/wdog.h>
+#include <nuttx/ioexpander/gpio.h>
+
+#include <arch/board/board.h>
+
+#include "chip.h"
+#include "tlsr82_gpio.h"
+#include "tlsr8278adk80d.h"
+
+#if defined(CONFIG_DEV_GPIO) && !defined(CONFIG_GPIO_LOWER_HALF)
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+struct tlsr82gpio_dev_s
+{
+  struct gpio_dev_s gpio;
+  pin_interrupt_t callback;
+  uint8_t id;
+};
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static int tlsr82_go_read(FAR struct gpio_dev_s *dev, FAR bool *value);
+static int tlsr82_go_write(FAR struct gpio_dev_s *dev, bool value);
+static int tlsr82_go_attach(FAR struct gpio_dev_s *dev,
+                            pin_interrupt_t callback);
+static int tlsr82_go_enable(FAR struct gpio_dev_s *dev, bool enable);
+static int tlsr82_go_setpintype(FAR struct gpio_dev_s *dev,
+                                enum gpio_pintype_e pintype);
+static int tlsr82_go_interrupt(int irq, void *context, void *arg);
```

Comment:
Probably do not need `FAR` for this arch in board files as well

Reply:
Done.